src/device/common.h

```diff
@@ -105,23 +105,25 @@
 #endif
 
 struct ncclShmemGroup {
-  ncclConnInfo *recvConns[NCCL_MAX_NVLS_ARITY];
-  ncclConnInfo *sendConns[NCCL_MAX_NVLS_ARITY];
-  void* srcs[NCCL_MAX_NVLS_ARITY+1];
-  void* dsts[NCCL_MAX_NVLS_ARITY+1];
+  ncclConnInfo *recvConns[NCCL_MAX_ARITY];
+  ncclConnInfo *sendConns[NCCL_MAX_ARITY];
+  void* userInput;
+  void* userOutput;
```

Comment:
`userBuff`, `RoleInput`, and `RoleOutput` are removed from prims_simple.h, and `userInput` and `userOutput` are now used instead.

Reply:
should move to latest NCCL on these fields.